autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/TabularColumnTransformer.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any, Dict, Optional, Union
+
+import numpy as np
+
+from sklearn.compose import ColumnTransformer, make_column_transformer
+from sklearn.pipeline import make_pipeline
+
+import torch
+
+from autoPyTorch.pipeline.components.preprocessing.tabular_preprocessing.base_tabular_preprocessing import (
+    autoPyTorchTabularPreprocessingComponent
+)
+from autoPyTorch.pipeline.components.preprocessing.tabular_preprocessing.utils import get_tabular_preprocessers
+
+
+class TabularColumnTransformer(autoPyTorchTabularPreprocessingComponent):
+    def __init__(self, random_state: Optional[Union[np.random.RandomState, int]] = None):
+        super().__init__()
+        self.random_state = random_state
+        self.column_transformer: Optional[ColumnTransformer] = None
+
+    def get_column_transformer(self) -> ColumnTransformer:
+        """
+        Get fitted column transformer that is wrapped around
+        the sklearn preprocessor. Can only be called if fit()
+        has been called on the object.
+        Returns:
+            BaseEstimator: Fitted sklearn column transformer
+        """
+        if self.column_transformer is None:
+            raise AttributeError("{} can't return column transformer before transform is called"
+                                 .format(self.__class__.__name__))
+        return self.column_transformer
+
+    def fit(self, X: Dict[str, Any], y: Any = None) -> "TabularColumnTransformer":
+        """
+        Creates a column transformer for the chosen tabular
+        preprocessors
+        Args:
+            X (Dict[str, Any]): fit dictionary
+
+        Returns:
+            "TabularColumnTransformer": an instance of self
+        """
+        self.check_requirements(X, y)
+
+        preprocessors = get_tabular_preprocessers(X)
+        numerical_pipeline = make_pipeline(*preprocessors['numerical'])
```

Comment:
If the preprocessors['numerical'] is empty, because the data is only categorical, will this still work?

Reply:
I thought I made it work but apparently it doesn't so I will fix it and add a test to ensure only numerical and only categorical cases work